### DISCUSSION TITLE
Add integrity attributes to check security of CDN resources

### DISCUSSION THREAD

**Discussion by domi-nika60** [swagger-api/swagger-codegen#11163] [Q&A]
Hi,

I'm wondering if there is any way to add Subresource Integrity attributes to resources of swagger codegen like 
`https://unpkg.com/swagger-ui-dist/swagger-ui.css` or `https://unpkg.com/swagger-ui-dist/swagger-ui-bundle.js`?

Generated openApi Swagger documentation consists:
```
<head>
    <meta charset='UTF-8' />
    <title>my title</title>
    <link rel='icon' type='image/png' href='https://unpkg.com/swagger-ui-dist/favicon-32x32.png' sizes='32x32' />
    <link rel='icon' type='image/png' href='https://unpkg.com/swagger-ui-dist/favicon-16x16.png' sizes='16x16' />
    <script src='https://unpkg.com/swagger-ui-dist/swagger-ui-bundle.js'></script>
    <script src='https://unpkg.com/swagger-ui-dist/swagger-ui-standalone-preset.js'></script>
    <link rel='stylesheet' type='text/css' href='https://unpkg.com/swagger-ui-dist/swagger-ui.css' />
    <link rel='stylesheet' type='text/css' href='https://unpkg.com/swagger-ui-themes@3.0.0/themes/3.x/theme-flattop.css' />
</head>
```

How can I add `integrity` attribute to use base64-encoded cryptographic hash to ensure secure resource usage and prevent data manipulation attacks?
I need some mechanism which ensures the actual included content matches the expected content 
(like Subresource Integrity [https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity))

**Comment by pskowronek** [comment]
anyone? :)

**Comment by LoganB99** [comment]
@daniel-kmiecik @djankows @ewaostrowska

This is a Security Weakness
CWE CWE-693
OWASP A1 Broken Access Control

**Comment by LoganB99** [comment]
created https://github.com/swagger-api/swagger-codegen/issues/12590
